apps/costcontrol/js/message_handler.js

```diff
@@ -323,26 +272,74 @@
     });
   }
 
+  function getCallback(activity, app) {
```

Comment:
Please, rename `getCallback` to precise a little bit more what this function is doing and reame `activity` to `type` to reflect what are you passing as first parameter.